docs/HowToFAQ/FAQFunctionByMethodProof.md

```diff
@@ -0,0 +1,87 @@
+---
+title: I can't prove the equivalence between the method part of a `function by method`` and the function itself
+---
+
+## Question
+
+I can't prove the equivalence between the method part of a function by method and the function itself.
+It seems that my invariants can't be maintained by the loop, and I don't know how to prove them.
+
+## Answer
+
+Consider the simple problem of computing the sum of a sequence of integers.
+You might be tempted to immediately write the following equivalence,
+with your best guess of an invariant
+and a hint at the end to help Dafny realize that the result is correct.
+However you cannot prove the invariant you just wrote.
+
+```
+{% include_relative FAQFunctionByMethodProof1.dfy %}
+```
+
+Let's unroll the computation of the function on a sequence of 3 elements.
+The function will compute:
+
+`s[0] + (s[1] + (s[2] + 0))`
+
+But the loop will compute
+
+`((0 + s[0]) + s[1]) + s[2]`
+
+In the function, we add the first element to the result of computing the sum of the remaining elements.
```

Comment:
I would indicate that the first element is `s[0]`, the "sum of the remaining elements" is `s[1] + (s[2] + 0)`, the last element is `s[2]`, and the "sum of the initial terms" is `(0 + s[0]) + s[1]` so that the reader can connect the text explanation to the source code.